OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -150,7 +150,7 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 
 				var minefield = GetMinefieldCells(minefieldStart, cell, Info.MinefieldDepth)
 					.Where(c => IsCellAcceptable(self, c) && self.Owner.Shroud.IsExplored(c)
-						&& movement.CanEnterCell(c, null, BlockedByActor.Immovable) && (mobile != null && mobile.CanStayInCell(c)))
+						&& movement.CanEnterCell(c, null, BlockedByActor.Immovable) && mobile?.CanStayInCell(c) == true)
```

Comment:
```suggestion
						&& movement.CanEnterCell(c, null, BlockedByActor.Immovable) && mobile?.CanStayInCell(c))
```
(Commit "Use pattern matching")

Reply:
I'm afraid we need that

![image](https://user-images.githubusercontent.com/11148519/167621337-bb79156a-6a03-49d3-afe9-28b761f5d099.png)

Reply:
IMO `(mobile != null && mobile.CanStayInCell(c))` is clearer than `mobile?.CanStayInCell(c) == true` due to this exact confusion.

Reply:
I too prefer the old variant over having `== true`. Can you change it back please?

Reply:
A compromise could be `mobile?.CanStayInCell(c) ?? false` if reverting this change means having to relax one of the new style rules.

Reply:
I finally went for the original solution